include/extractor/guidance/validation_handler.hpp

```diff
@@ -177,6 +179,81 @@ class ValidationHandler final : public IntersectionHandler
             }
         }
     }
+
+    void checkForSharpDualCarriagewayUturns(const NodeID from_nid,
+                                            const EdgeID via_eid,
+                                            const Intersection &intersection) const
+    {
+        // Detect situations where dual-carriageway fans-in from two ways into one and
+        // the Uturn from a dual-carriageway onto the dual-carriageway should be possible.
+        //
+        // . .
+        //     > .
+        // . .
+        //
+
+        // Todo: are there dual-carriageway intersections of larger size?
+        if (intersection.size() != 3)
```

Comment:
This prevents checks for any intersections. It only finds turns around traffic islands. Most of these will split next to intersections -> intersection.size() > 3